onnxruntime/python/tools/quantization/onnx_quantizer.py

```diff
@@ -694,10 +694,12 @@ def quantize_inputs(self, node, indices, initializer_use_weight_qType=True, redu
 
         return (quantized_input_names, zero_point_names, scale_names, nodes)
 
-    def quantize_weight(self, weight, qType, reduce_range=False):
+    def quantize_weight(self, weight, qType, reduce_range=False, keep_float_weight=False):
```

Comment:
what's the reason to keep float here?

Reply:
If keep_float_weight is True, then we don't want to quantize the weight.
By reading the previous code, it shows that it always quantize the weight and make the 'quantized' weight as the input of DequanlizeLinear node. But for TRT QDQ, we need a complete QDQ pair and original weight.

Reply:
> what's the reason to keep float here?

TRT8 doesn't support pre-quantized INT8 weights. The pattern must be floating point weights + Q + DQ + op. That's why we need to keep floating point weights and have a whole QDQ pair after it.